File: packages/js-sdk/src/types/company.ts
/**
 * TypeScript types for TurboQuote — Company entity and request/response types
 */

import type { PaginationParams, PaginatedResponse } from './quote-shared';

// ============================================
// DOMAIN TYPES
// ============================================

export interface Company {
  id: string;
  orgId: string;
  name: string;
  phone: string | null;
  city: string | null;
  state: string | null;
  country: string | null;
  industryId: string | null;
  lastActivityDate: string | null;
  isActive: boolean;
  createdBy: string | null;
  createdOn: string;
  updatedOn: string;
}

// ============================================
// REQUEST TYPES
// ============================================

export interface CreateCompanyContactInput {
  name: string;
  email: string;
  phone?: string;
  title?: string;
}

export interface CreateCompanyRequest {
  name: string;
  contacts: CreateCompanyContactInput[];
  phone?: string;
  city?: string;
  state?: string;
  country?: string;
  industryId?: string;
}

export interface UpdateCompanyRequest {
  name?: string;
  phone?: string;
  city?: string;
  state?: string;
  country?: string;
  industryId?: string;
}

export interface ListCompaniesOptions extends PaginationParams {
  industryIds?: string | string[];
}

// ============================================
// RESPONSE TYPES
// ============================================

export type CompanyListResponse = PaginatedResponse<Company>;

File: packages/py-sdk/src/turbodocx_sdk/types/company.py
"""
TypedDict definitions for TurboQuote -- Company entity.
"""

from __future__ import annotations

from typing import Any, Dict, List, Optional
from typing_extensions import TypedDict

from .quote_type import QuoteType


class Company(TypedDict, total=False):
    id: str
    orgId: str
    name: str
    phone: Optional[str]
    city: Optional[str]
    state: Optional[str]
    country: Optional[str]
    industryId: Optional[str]
    lastActivityDate: Optional[str]
    isActive: bool
    createdBy: Optional[str]
    createdOn: str
    updatedOn: str
    contactCount: int
    industry: QuoteType


class CreateCompanyContactInput(TypedDict, total=False):
    name: str
    email: str
    phone: Optional[str]
    title: Optional[str]


class _CreateCompanyRequestRequired(TypedDict):
    name: str
    contacts: List[CreateCompanyContactInput]


class CreateCompanyRequest(_CreateCompanyRequestRequired, total=False):
    phone: Optional[str]
    city: Optional[str]
    state: Optional[str]
    country: Optional[str]
    industryId: Optional[str]


class UpdateCompanyRequest(TypedDict, total=False):
    name: str
    phone: Optional[str]
    city: Optional[str]
    state: Optional[str]
    country: Optional[str]
    industryId: Optional[str]


class ListCompaniesOptions(TypedDict, total=False):
    limit: int
    offset: int
    query: str
    industryIds: Any  # string | string[]


class CompanyListResponse(TypedDict):
    results: List[Company]
    totalRecords: int
